spec/controllers/api/v1/marketing/lists_controller_spec.rb

```diff
@@ -0,0 +1,250 @@
+# bundle exec rspec --format documentation spec/controllers/api/v1/marketing/lists_controller_spec.rb 
+
+require 'rails_helper'
+
+RSpec.describe Api::V1::Marketing::ListsController do
+
+    # 
+    # initialize()
+    # 
+    def initialize (x)
+        @base_earl = 'api/marketing/lists?'
+        @bugger = false
+    end
+
+    # 
+    # show_test()
+    # 
+    def show_test (t, p)
+        puts ''
+        puts '  trying ' + @base_earl + p.to_query
+        puts '    [' + t + ']'
+    end
+
+    # 
+    # show_subtest()
+    # 
+    def show_subtest (message, result = true)
+        printf "    %-80s %s\n", message, (result ? '[OK]' : '*** FAILED ***')
+    end
+
+    # 
+    # bugger_add()
+    # 
+    def bugger_add (message)
+        pp message if @bugger
+    end
+
+    # 
+    # describe
+    # 
+    describe 'GET get_list' do
+
+        # get with no params (code 1, method not allowed)
+        t1 = 'get with no params (code 1, method not allowed)'
+        it t1 do
+            params = {}
+            get 'get_list'
+            show_test t1, params
+            json = JSON.parse(response.body)
+            expect(response).to be_success
+            show_subtest 'response be_success'
+            expect(response).to have_http_status(200)
```

Comment:
I just meant that, in case of a failure, lines 53-57 will not be executed. But I guess their purpose is to log more information in case of a success (and to check the value of json[code])...